### ISSUE TITLE
what is the canonical way to use (production) webview2 for winstore focused web apps without getting tripped up by CORS?

### ISSUE THREAD

**Issue by diffent** [MicrosoftEdge/WebView2Feedback#3148]
looking around online i found the template to set up a webview2 C# app for a UWP app.  good.  it shows bing.com site in it.

now what if we want to have a web app (local files in-app or from an ordinary https:// server), and/or want to make fetch() calls 
in the javascript of the web app without running into CORS issues.  what is the recommended way to set this up?

let's assume for now the app is a pure JS  web app (e.g. little or no interaction w/ C# code except as a webview2 wrapper)

(that being the easiest to get going, in theory)

i looked around online and there were a variety of suggestions, but it seems they may be for pre-release code
since the examples i found don't compile (noted below)

we would rather not do this as a guessing game, but do it the recommended MS way.  it is difficult to find proper solutions
online because one does not know if the examples are still valid w/ production webview2.

this was a suggestion allow file:// and or bypass cors issues when doing fetch( ) in the JS

// but some of the MS doc says the first call requires 4 args?  and it doesnt compile with 1 arg or 4 args (even though the documentation says that args 2,3,4 are optional)... 

var op = new CoreWebView2EnvironmentOptions("--disable-web-security"); 
var env = await CoreWebView2Environment.CreateAsync(null, null, op);
await webView.EnsureCoreWebView2Async(env);

**Comment by jasonstephen15** [comment]
@diffent - can you explain your app scenario in which you are reaching CORS limit or experiencing issues with CORS?

**Comment by diffent** [comment]
thanks for the response, i figured out a solution via some trial and error and help from other posts and MS staff.  i will try to push up a sample solution soon.  the app made it to the store!  i did not need the environment settings per above.  here is the rough idea:

this is from a webview2 winui2 example w/ C# wrapper to build a UWP app for the MS app store

use case of : main web app html and js are built in to the app package in a subdir of Assets dir in the project

older solutions online that say to use ms-appx:// or similar non- http(s):// url prefixes to get to files in local app Assets folder
do not seem to work any more with WebView2, at least in the context of UWP WinUI2

```
public sealed partial class MainPage : Page
    {
        public MainPage()
        {
           
            this.InitializeComponent();

            var result = this.SetUpWebView(); // not using result but who cares, it is to get rid of warning

        }
        public async Task SetUpWebView()
        {
            await this.webView.EnsureCoreWebView2Async(); // ensure the CoreWebView2 is created

            var core_wv2 = webView.CoreWebView2;
            if (core_wv2 != null)
            {
                // appnameshort can be an abbrev for your app name
		// note that older examples may have lower case "assets" folder in the project
                core_wv2.SetVirtualHostNameToFolderMapping(
                    "appnameshort", "Assets",
                    Microsoft.Web.WebView2.Core.CoreWebView2HostResourceAccessKind.Allow);

		// folderInAssets can be any folder name in your Assets folder where you will put your html/js etc directory tree
                // use http: here until you get https: on backend API server or fetch( ) will not work in javascript run by webview2 
                webView.Source = new Uri("http://appnameshort/folderInAssets/mainapp.html");
            }

        }
    }
```

other useful hint i found that is not obvious to a new-comer to WebView2:

use ctrl shift i when cursor is on the web view in the running app to show javascript dev tools & javascript console
(not sure yet if cursor needs to be on the web view but couldnt hurt, right?)

you may need to install webview2 dev tools in visual studio first, there is some documentation for that
somewhere online but I forgot where at the moment (maybe MS staff can clarify?)

**Comment by jasonstephen15** [comment]
@diffent - Glad to hear your scenario is working. You don't need to install devtools in visual studio, it comes enabled with webview2. We have some documentation about debugging webview2, that has the advice of using ctr shift i: https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/debug-devtools?tabs=dotnetcsharp.

I've opened up a doc item to make sure we document any folder mapping quirks with UWP. Closing post for now thanks!
